### PR TITLE
Add reviewers to a PR (WIP)

### PR DESCRIPTION
**Pull request by bobrowadam**

### CODE REVIEW

File: magithub-user.el [hunk 1]
```diff
@@ -45,6 +45,12 @@
     (define-key m [remap magit-delete-thing] #'magithub-assignee-remove)
     m))
 
+(defvar magit-magithub-reviewer-section-map
+  (let ((m (make-sparse-keymap)))
+    (set-keymap-parent m magit-magithub-user-section-map)
+    (define-key m "a" #'magithub-reviewer-add)
```
Comment (vermiculus) [inline magithub-user.el]: Remap `magithub-add-thing` instead :)

File: magithub-user.el [hunk 2]
```diff
@@ -123,6 +155,14 @@
    (lambda (user) (let-alist user .login))
    nil t default-user))
 
+(defun magithub-user-choose-reviewer (prompt &optional repo default-user)
```
[the remark this answers is not in the document]
Comment (bobrowadam) [reply magithub-user.el]: Yea you are right. I think we can make it mandatory.

File: magithub-user.el (hunk 1 above)
Comment (bobrowadam) [reply magithub-user.el]: Will do :)

**Comment by bobrowadam** [comment]
@vermiculus Any suggestions on where should we get the reviewers list when updating the issue view? I noticed that the reviewers are not part of the issue.

**Comment by bobrowadam** [comment]
@vermiculus 
`magithub-reviewer-add` seems to work. I'm having issues with `magithub-reviewer-remove` though. When pointing on the reviewers names the value I get for `(thing-at-point 'github-user)` is nil.
Might be related to the following error:
`Invalid face reference: magit-user`
Any idea what may cause that?

**Comment by vermiculus** [comment]
I can look into it locally.

**Comment by bobrowadam** [comment]
Fetching the collaborators on each refresh is too slow. I will have to think of a better way.

**Comment by vermiculus** [comment]
Cache them. Use the :issues bucket.

**Comment by vermiculus** [comment]
Now that I'm finally at a computer again with some time, I want to expand on my previous comment. There's a caching system present in `magithub-core.el` that exposes function `magithub-cache`. You can use the definition of `magithub-user-me` as a guide.

**Comment by bobrowadam** [comment]
Great, I was planning to implement that tomorrow. Also, let me know if you have any idea why would the reviewers object is 'nil' when applying `reviewer-remove`.

**Comment by vermiculus** [comment]
I'd like to experiment with it, but my test repository has no collaborators – I sent you an invite: https://github.com/vermiculus/my-new-repository/invitations

**Comment by vermiculus** [comment]
I found the problem anyway :-) you're passing the user's ID; not the review ID. What will probably need to happen is to search through the open reviews on the PR and find a match on the username – then delete that.

**Comment by vermiculus** [comment]
Also, if you rebase onto `master`, that should resolve the issues you're having with CI.

**Comment by bobrowadam** [comment]
1. Added caching for `reviewers`, hope I got it right.
2. I was following [`#delete-a-review-request`](https://developer.github.com/v3/pulls/review_requests/#delete-a-review-request)
This API only needs a list of names for deletion. I tested it by using curl and it seems to do the job.
3. Unless I'm doing anything wrong, I'm already rebased over master.

**Comment by vermiculus** [comment]
Some commits have been made since the time your forked – one of those commits fixed the tests that check for clean compile, etc.

**Comment by vermiculus** [comment]
Looks like there are some compile errors: https://travis-ci.org/vermiculus/magithub/jobs/432728627#L996-L997

```
magithub-issue.el:422:1:Warning: Unused lexical variable ‘assignee’
magithub-issue.el:429:69:Error: assignment to free variable ‘reviewer’
```

**Comment by bobrowadam** [comment]
@vermiculus I think that those errors should be fixed now. The only issue is with ghub's version it seems:
https://travis-ci.org/vermiculus/magithub/jobs/444470385#L1024

**Comment by vermiculus** [comment]
There are a few other changes to make (typos and other stuff) but I'll push those, squash, and merge later tonight.

**Comment by vermiculus** [comment]
(Thanks!!)

**Comment by vermiculus** [comment]
I think the above issue is due to the user's login/ID being passed as the review ID. These are not the same values.

**Comment by vermiculus** [comment]
I think I've fixed it – tested that I could add+remove you from vermiculus/my-new-repository#6. Can you test as well to make sure I didn't break something that was working for you before?

**Comment by bobrowadam** [comment]
@vermiculus any updates?

**Comment by vermiculus** [comment]
I'm sorry I haven't been able to focus on this lately :( hopefully I'll get some time during the thanksgiving weekend

**Comment by bobrowadam** [comment]
Sure don't worry about it :) I was just wondering if you need anything more from me to do

**Pull request closed without merging**